examples/cifar10/readme.md

```diff
@@ -65,22 +65,28 @@ using namespace tiny_dnn::activation;
 
 template <typename N>
 void construct_net(N& nn) {
-    typedef convolutional_layer<activation::identity> conv;
-    typedef max_pooling_layer<relu> pool;
-
-    const int n_fmaps = 32; ///< number of feature maps for upper layer
-    const int n_fmaps2 = 64; ///< number of feature maps for lower layer
-    const int n_fc = 64; ///< number of hidden units in fully-connected layer
-
-    nn << conv(32, 32, 5, 3, n_fmaps, padding::same)
-        << pool(32, 32, n_fmaps, 2)
-        << conv(16, 16, 5, n_fmaps, n_fmaps, padding::same)
-        << pool(16, 16, n_fmaps, 2)
-        << conv(8, 8, 5, n_fmaps, n_fmaps2, padding::same)
-        << pool(8, 8, n_fmaps2, 2)
-        << fully_connected_layer(4 * 4 * n_fmaps2, n_fc)
-        << fully_connected_layer(n_fc, 10)
-        << softmax_layer(10);
+    using conv = convolutional_layer;
+    using pool = max_pooling_layer<identity>;
+    using fc = fully_connected_layer;
+    using relu = relu_layer;
+    using softmax = softmax_layer;
+
+  const serial_size_t n_fmaps  = 32;  // number of feature maps for upper layer
+  const serial_size_t n_fmaps2 = 64;  // number of feature maps for lower layer
+  const serial_size_t n_fc     = 64;  // number of hidden units in fc layer
+
+  nn << conv(32, 32, 5, 3, n_fmaps, padding::same)        // C1
+     << pool(32, 32, n_fmaps, 2)                          // P2
+     << relu(16, 16, n_fmaps)                             // activation
+     << conv(16, 16, 5, n_fmaps, n_fmaps, padding::same)  // C3
+     << pool(16, 16, n_fmaps, 2)                          // P4
+     << relu(8, 8, n_fmaps)                               // activation
+     << conv(8, 8, 5, n_fmaps, n_fmaps2, padding::same)   // C5
+     << pool(8, 8, n_fmaps2, 2)                           // P6
+     << relu(4, 4, n_fmaps)                               // activation
+     << fc(4 * 4 * n_fmaps2, n_fc)                        // FC7
+     << relu(n_fc)                                        // activation
+     << fc(n_fc, 10) << softmax(10);                      // softmax
```

Comment:
fully_connected is a high level interface name, but the underneath operation it's just a linear matrix multiplication.